psyneulink/core/compositions/composition.py

```diff
@@ -9037,16 +9204,19 @@ def run(
                         )
                     node.initialize(initialize_cycle_values[node], context)
 
-        if reset_stateful_functions_to is None:
-            reset_stateful_functions_to = {}
-
-        for node in reset_stateful_functions_to:
-            # make sure the args to reset are in the form of an iterable so they can be unpacked
-            try:
-                iter(reset_stateful_functions_to[node])
-            except TypeError:
-                reset_stateful_functions_to[node] = [reset_stateful_functions_to[node]]
-            node.reset(*reset_stateful_functions_to[node], context=context)
+        try:
+            for node in reset_stateful_functions_to:
+                # make sure the args to reset are in the form of an iterable so they can be unpacked
+                try:
+                    iter(reset_stateful_functions_to[node])
+                except TypeError:
+                    reset_stateful_functions_to[node] = [reset_stateful_functions_to[node]]
+                if (isinstance(reset_stateful_functions_when, Never) or
+                        node not in reset_stateful_functions_when) and \
+                        isinstance(node.reset_stateful_function_when, Never):
+                    node.reset(*reset_stateful_functions_to[node], context=context)
+        except TypeError:
```

Comment:
should the entire loop exit if one iterations raises TypeError?

Reply:
This was previously being used to catch `TypeError`s arising from when `reset_stateful_functions_to` was `None`, but the block was restructured in ffad3efb0747fedd8306bad83a2bfc5dd5ddee29 and no longer does this.